packages/ra-core/src/dataProvider/useGetMany.ts

```diff
@@ -77,39 +79,61 @@ export const useGetMany = <RecordType extends RaRecord = any>(
                 .getMany<RecordType>(resource, { ids, meta })
                 .then(({ data }) => data);
         },
-        {
-            placeholderData: () => {
-                const records =
-                    !ids || ids.length === 0
-                        ? []
-                        : ids.map(id => {
-                              const queryHash = hashQueryKey([
-                                  resource,
-                                  'getOne',
-                                  { id: String(id), meta },
-                              ]);
-                              return queryCache.get<RecordType>(queryHash)
-                                  ?.state?.data;
-                          });
-                if (records.some(record => record === undefined)) {
-                    return undefined;
-                } else {
-                    return records as RecordType[];
-                }
-            },
-            onSuccess: data => {
-                // optimistically populate the getOne cache
-                data.forEach(record => {
-                    queryClient.setQueryData(
-                        [resource, 'getOne', { id: String(record.id), meta }],
-                        oldRecord => oldRecord ?? record
-                    );
-                });
-            },
-            retry: false,
-            ...options,
+        placeholderData: () => {
+            const records =
+                !ids || ids.length === 0
+                    ? []
+                    : ids.map(id => {
+                          const queryHash = hashKey([
+                              resource,
+                              'getOne',
+                              { id: String(id), meta },
+                          ]);
+                          return queryCache.get<RecordType>(queryHash)?.state
+                              ?.data;
+                      });
+            if (records.some(record => record === undefined)) {
+                return undefined;
+            } else {
+                return records as RecordType[];
+            }
+        },
+        retry: false,
+        ...queryOptions,
+    });
+
+    useEffect(() => {
+        if (result.data != null) {
```

Comment:
why two imbricated ifs instead of just one with `&`? Same for other hooks.

Reply:
What imbricated if?